lib/matplotlib/tests/test_axes.py

```diff
@@ -1478,6 +1478,20 @@ def test_pcolorargs():
                       match='are not monotonically increasing or decreasing'):
         ax.pcolormesh(X, Y, Z, shading='auto')
 
+    # GH 26093
```

Comment:
Hi @oscargus,

I add a method to check if the mask array is unwritable in `Axes` class and I don't know if it's proper since the method is irrelevant to what the `Axes` class is used for.

https://github.com/Rylie-W/matplotlib/blob/b1d657c663e451d06bfbf8e454e40b55eb4b5d8c/lib/matplotlib/axes/_axes.py#L5743-L5744

Reply:
I think that it may be better to just copy it locally. (Two statements, but still only one line.)

I can confirm that you were right about that `mask` is read-only (which you probably knew, but I didn't). Still, for understanding the test, I think that the `assert mask.flags.writeable is False` just after mask is defined makes sense. (Unless it breaks depending on NumPy version...)

Reply:
Does the current implementation make sense? I hope I understand it correctly.

Reply:
Test for sure.

I sort of liked the logic to check if one should copy, not just as a method, but I guess always copying is probably OK as well.

Reply:
I am too late, but I think it is right that we still copy if the mask is writable.  I do not think we should be writing back to the user’s input array.